protocol/messenger_handler.go

```diff
@@ -1541,6 +1541,19 @@ func (m *Messenger) HandleDeleteMessage(state *ReceivedMessageState, deleteMessa
 				return ErrInvalidDeletePermission
 			}
 		}
+
+		if chat.ChatType == ChatTypePrivateGroupChat || chat.ChatType == ChatTypePublic {
```

Comment:
Would be good to get some test coverage for the changes in `HandleDeleteMessage`.